### ISSUE TITLE
When pasting a text in tan app on a secondary monitor, the text goes to the app on the main monitor

### ISSUE THREAD

**Issue by soberano** [p0deje/Maccy#691]
### Before Submitting Your Bug Report

- [X] I have verified that there isn't already an issue reporting the same bug to prevent duplication.
- [X] I have seen the [FAQ](https://github.com/p0deje/Maccy?tab=readme-ov-file#faq).

### Maccy Version (see 'About' window)

0.29.2 (26)

### macOS Version

14.1.2 (23B92)

### Maccy Settings

```Shell
{
    "KeyboardShortcuts_delete" = "{\\"carbonKeyCode\\":51,\\"carbonModifiers\\":2048}";
    "KeyboardShortcuts_pin" = "{\\"carbonKeyCode\\":35,\\"carbonModifiers\\":2048}";
    "KeyboardShortcuts_popup" = "{\\"carbonKeyCode\\":8,\\"carbonModifiers\\":768}";
    "LaunchAtLogin__hasMigrated" = 1;
    "NSWindow Frame SUStatusFrame" = "2851 698 400 134 1800 89 1920 1055 ";
    "NSWindow Frame SUUpdateAlert" = "2450 581 620 398 1800 89 1920 1055 ";
    "NSWindow Frame com.sindresorhus.Settings.FrameAutosaveName" = "629 652 442 322 0 0 1800 1125 ";
    SUEnableAutomaticChecks = 1;
    SUHasLaunchedBefore = 1;
    SULastCheckTime = "2023-12-05 11:19:29 +0000";
    SUUpdateGroupIdentifier = 1532164247;
    WebKitDefaultFontSize = 13;
    WebKitJavaScriptEnabled = 0;
    WebKitPluginsEnabled = 0;
    WebKitStandardFont = "-apple-system-font";
    WebKitUserStyleSheetEnabledPreferenceKey = 1;
    WebKitUserStyleSheetLocationPreferenceKey = "/Applications/Maccy.app/Contents/Frameworks/Sparkle.framework/Resources/ReleaseNotesColorStyle.css";
    enabledPasteboardTypes =     (
        "public.html",
        "public.png",
        "public.rtf",
        "public.tiff",
        "public.utf8-plain-text",
        "public.file-url"
    );
    ignoredPasteboardTypes =     (
        "com.agilebits.onepassword",
        "de.petermaurer.TransientPasteboardType",
        "net.antelle.keeweb",
        "Pasteboard generator type",
        "com.typeit4me.clipping"
    );
    migrations =     {
        "2020-04-25-allow-custom-ignored-types" = 1;
        "2020-06-19-use-keyboardshortcuts" = 1;
        "2020-09-01-ignore-keeweb" = 1;
        "2021-02-20-allow-to-customize-supported-types" = 1;
        "2021-06-28-add-title-to-history-item" = 1;
        "2021-10-16-remove-dynamic-pasteboard-types" = 1;
        "2022-08-01-rename-suppress-clear-alert" = 1;
        "2022-11-14-add-html-rtf-to-supported-types" = 1;
        "2023-01-22-add-regexp-search-mode" = 1;
    };
    pasteByDefault = 1;
    previewDelay = 1000;
}
```

### Description

When running two windows of the same application in different monitors (built-in mac and external) and trying to paste a text to an application on a secondary monitor, the text goes to the same application on the main monitor.

For example, if I have two terminal windows on two different monitors, and I try to paste into the terminal window on the secondary monitor, the text is being pasted into the terminal window on the main monitor.
This happens in other applications as well, such as Google Chrome and Microsoft Excel.

### Steps to Reproduce

Open 2 windows of the same application in main and secondary monitors, for example iTerm
Copy a text into clipboard
Focus on the instance on the secondary monitor
Press Command-Shift-C and select an entry to paste
Pasted text goes to the application instance in the main monitor (built-in monitor in my case) and the focus switches to it as well.

If both app windows are on the same monitor (main or secondary) the the pasted text goes to the focused instance as expected.

**Comment by p0deje** [comment]
I cannot reproduce this on macOS 14.3 with iTerm having the following displays:

<img width="476" alt="Untitled" src="https://github.com/p0deje/Maccy/assets/665846/bdbd53d3-8446-4956-9991-d0a44e4f9451">
<img width="581" alt="Untitled1" src="https://github.com/p0deje/Maccy/assets/665846/24ce6cbb-dce4-4070-9435-486222bd09a8">

Having 2 iTerm windows on separate screens, paste always goes to the last focused one. 

Please try on the latest macOS and re-open if it doesn't work still.

**Comment by dszymczuk** [comment]
Hi.

I have the same issue. I have opened Visual Studio Code on the 1st (built-in) and 2nd monitor (external).
When I tried to paste in VSC on the external monitor, the text was pasted in VSC on 1st monitor.

It's ultra annoying.

```
MacOS: Sonoma 14.5
Maccy: Version 1.0.0 (31)
```
My arrangement:

<img width="586" alt="Screenshot 2024-07-24 at 10 13 17" src="https://github.com/user-attachments/assets/3b287ff4-d92d-4c10-be69-524464f1f746">
